src/renderer/vt/VtSequences.cpp

```diff
@@ -266,15 +266,15 @@ using namespace Microsoft::Console::Render;
 [[nodiscard]] HRESULT VtEngine::_SetGraphicsRenditionRGBColor(const COLORREF color,
                                                               const bool fIsForeground) noexcept
 {
-    const std::string fmt = fIsForeground ?
-                                "\x1b[38;2;%d;%d;%dm" :
-                                "\x1b[48;2;%d;%d;%dm";
-
     DWORD const r = GetRValue(color);
     DWORD const g = GetGValue(color);
     DWORD const b = GetBValue(color);
 
-    return _WriteFormattedString(&fmt, r, g, b);
+    // An example string with max length would be "\x1b[38;2;128;128;128m", which has length = 19.
+    char buf[19];
+    char* end = fmt::format_to(std::begin(buf), FMT_COMPILE("\x1b[{};2;{};{};{}m"), fIsForeground ? 38 : 48, r, g, b);
+
+    return _Write({ buf, static_cast<std::string_view::size_type>(end - buf) });

```

Comment:
If you don't want to incur any allocation overhead here, you could use some variant of std::pmr::string with a memory resource that allocates on the stack. But we don't have that yet. What we do have already is boost!

```cpp
// that would have to put into the pch file
#include <boost/container/small_vector.hpp>

// An example string with max length would be "\x1b[38;2;128;128;128m", which has length = 19 + \0.
boost::container::small_vector<char, 20> buf;
fmt::format_to(std::back_inserter(buf), FMT_COMPILE("\x1b[{};2;{};{};{}m"), fIsForeground ? 38 : 48, r, g, b);
return _Write({ buf.data(), buf.size() });
```